### ISSUE TITLE
drawText location incorrect on scaled Canvas (CentOS 7 and Ubuntu 16)

### ISSUE THREAD

**Issue by mvitale** [Automattic/node-canvas#1037]
## Issue

This is a special case of #903 . I'm opening a new issue since I can't reopen that one. Feel free to close if this is the wrong way to surface the problem! 

I'm having the same issue on CentOS 7, and @zbjornson commented saying that he has encountered it on Ubuntu 16. See #903 for the results of some preliminary investigation.

## Your Environment
* Version of node-canvas: 1.6.7
* Environment: node v6.11.5 on CentOS 7

**Comment by chearon** [comment]
The numbers posted in #903 stand out to me because those numbers shouldn't be changing based on Cairo's transform matrix (or at least, that's not what we want). And they _definitely_ shouldn't be different on different platforms.

I remember being confused a while back why we had to check the transform matrix at all in order to get the baseline offset, but I convinced myself it was right. I now think that part of the code is actually compensating for a bug in Pango. You shouldn't need to do anything unusual to move a relative distance from the baseline to a scaled matrix, see [this](https://jsfiddle.net/vvu8duft/).

So as usual Pango is right on Linux but gets something wrong on macOS/Windows. And if I'm right, we can get around it by using a function on the layout instead, which seems more right to me anyways. I'll try a PR soon.
